### PR TITLE
[unit-tests-only] Added unit tests for shared with others view

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added unit tests for `SharedWithOthers` view

## Related Issue

- #5237 

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by pascalwengerter — DISMISSED** [review]

File: packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    it('should show the no content message component if the paginated resources is empty', () => {
+      const wrapper = getMountedWrapper()
+
+      expect(wrapper.find(listLoaderStub).exists()).toBeFalsy()
+      expect(wrapper.find(filesTableStub).exists()).toBeFalsy()
+      expect(wrapper.find(noContentStub).exists()).toBeTruthy()
+    })
+    describe('when length of paginated resources is greater than zero', () => {
+      const resourcesList = [
+        createFile({ id: '1234' }),
```
[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js:49]: I tried using the data from `sharedFiles` as resources input for the resource table. But sadly unit tests fail, see the log here https://gistpreview.github.io/?caee0de6f3f01f4ec4e3629c9107bc6b
### An item from `sharedFiles`
```
{
      id: '38',
      share_type: 3,
      uid_owner: 'admin',
      displayname_owner: 'admin',
      permissions: 1,
      stime: 1623242657,
      parent: null,
      expiration: null,
      token: 'sADy5cUdrz931MV',
      uid_file_owner: 'admin',
      displayname_file_owner: 'admin',
      additional_info_owner: null,
      additional_info_file_owner: null,
      path: '/Documents',
      mimetype: 'httpd/unix-directory',
      state: 0,
      storage_id: 'home::admin',
      storage: 2,
      item_type: 'folder',
      item_source: 3009,
      file_source: 3009,
      file_parent: 6,
      file_target: '/Documents',
      name: 'Quick action link',
      url: 'http://host.docker.internal:8080/s/sADy5cUdrz931MV',
      mail_send: 0,
      attributes: null,
      tags: [],
      sharedWith: [ { name: 'Quick action link', link: true } ]
 }
```
### output of `createFile` helper
```
 {
      id: 'file-id-1234',
      type: 'folder',
      status: 1,
      name: 'file-name-1234',
      path: '/file-path/1234',
      extension: '',
      share: { id: 'file-share-id-1234' },
      indicators: []
    }
```

[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js:49]: actually there are not enough information in the `sharedFile` data

[the remark this answers is not in the document]
Comment (pascalwengerter) [reply packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js:49]: Do you have an indication on which information is lacking and could you maybe add it? :)

[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js]: actually, these were the requirements of `OcResourceName` component from `OcResource`, I've updated the fixture file. Please have a look.

**Comment by phil-davis** [comment]
This PR was already 78 commits behind master. I rebased to get fresh CI.
@pascalwengerter and/or @kulmann please review.

File: packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pascalwengerter) [reply packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js:50]: I don't see the updated fixtures files, am I getting this wrong? 🤔

[the remark this answers is not in the document]
Comment (kiranparajuli589) [reply packages/web-app-files/tests/unit/views/SharedWithOthers.spec.js:50]: @pascalwengerter  i'm also surprised. I thought i'd fixed the code but seems like i didn't even pushed the updated code here in this branch. i'll quickly follow up on this ;)

**Comment by kiranparajuli589** [comment]
@pascalwengerter I've stubbed every `OC components` used inside the `ResourceTable` components. The fixtures don't need to be updated after that. please review again!

**Review by kulmann — APPROVED** [review]

**Pull request merged**